packages/terra-pills/src/FilterPills.jsx

```diff
@@ -0,0 +1,448 @@
+import React, {
+  useCallback, useLayoutEffect, useRef, useState,
+} from 'react';
+import {
+  KEY_BACK_SPACE,
+  KEY_DELETE,
+  KEY_END,
+  KEY_HOME,
+  KEY_LEFT,
+  KEY_RIGHT,
+} from 'keycode-js';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import PropTypes from 'prop-types';
+import ResizeObserver from 'resize-observer-polyfill';
+import ThemeContext from 'terra-theme-context';
+import { injectIntl } from 'react-intl';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
+import Pill from './private/_Pill';
+import PillsUtils from './private/PillsUtils';
+import RollUpPill from './private/_RollupPill';
+import styles from './private/Pill.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * The string that labels the collection of pills, used in cases where the text label is not visible on
+   * the screen and required for minimum accessibility standards. Providing this prop adds the `aria-label`
+   * attribute to the Pill List container element. (Required)
+   */
+  ariaLabel: PropTypes.string.isRequired,
```

Comment:
Should this be required? What if there is a visible label on the screen, could a screen reader end up reading double labels?

Reply:
Nevermind on the double label question, since I read the next prop description. But it does seem odd to make this a required prop when it won't be used if the `ariaLabelledBy` prop is provided.

Reply:
Making the `ariaLabel` required is the price of proper accessibility, even though teams should ideally use the `ariaLabelledBy` prop and provide their own styled and visible label. 

Since the filter pills can be used in a wide variety of places we might not expect or have planned for yet, we can't necessarily add a visible label with fixed styling as we normally would do, and providing the ariaLabel is a good fallback for if it is ever used in higher-order component constructions where the context is clear for visual users and the non-visible label will be appropriate and provide the context and clarity for the screen reader user. If anything, it will be a guaranteed fallback to meet A11Y success criteria for proper labeling.

Reply:
Marking as resolved since @jmsv6d is unavailable for the rest of this PR, an @neilpfeiffer's comment explains it.